tests/compiler/retain-i32.optimized.wat

```diff
@@ -27,4 +250,7 @@
   i32.load8_s
   drop
  )
+ (func $~start
+  call $start:retain-i32
+ )
```

Comment:
`retain-i32.optimized.wat` also pretty huge regression

Reply:
Might this have something to do with sign extension ops not being recognized by precompute or something?